### PR TITLE
fix: name for compound child tasks

### PR DESCRIPTION
**Pull request by prmukherj**
Compound child task_names() should have a format of the parent name followed by child number. Please see the test for details.

### CODE REVIEW

File: tests/test_new_meshing_workflow.py [hunk 1]
```diff
@@ -1507,3 +1505,47 @@ def test_switching_workflow_interface(new_mesh_session):
...
+    watertight.add_local_sizing.add_child_and_update(
+        state={
+            "boi_control_name": "outlet",
+            "boi_face_label_list": ["outlet"],
+            "boi_size": 10,
+        }
+    )
+
+    assert {
+        "child_1_of_add_local_sizing",
```
Comment (seanpearsonuk) [inline tests/test_new_meshing_workflow.py:1539]: Is there a reason for preferring child_n_of_ prefix over _child_n suffix?

Comment (prmukherj) [reply tests/test_new_meshing_workflow.py:1539]: No, is the latter better? We can update it.

Comment (seanpearsonuk) [reply tests/test_new_meshing_workflow.py:1539]: Does the _of_ part look a bit awkward in an object name? It's also a bit longer.

Comment (prmukherj) [reply tests/test_new_meshing_workflow.py:1539]: I agree. Will update the same.

**Review by seanpearsonuk — APPROVED** [review]

**Review by mkundu1 — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

**Pull request merged**